include/h2o/balancer.h

```diff
@@ -41,16 +41,38 @@ typedef struct st_h2o_balancer_callbacks_t {
     h2o_balancer_destroyer destroy;
 } h2o_balancer_callbacks_t;
 
+typedef enum en_h2o_balancer_type_t {
+    H2O_BALANCER_TYPE_ROUND_ROBIN,
+    H2O_BALANCER_TYPE_LEAST_CONN,
+    H2O_BALANCER_TYPE_HASH
+} h2o_balancer_type_t;
+
 struct st_h2o_balancer_t {
     const h2o_balancer_callbacks_t *callbacks;
+    const int idempotent;
+    h2o_balancer_type_t type;
 };
 
+typedef enum en_h2o_balancer_hash_key_type_t {
+    H2O_BALANCER_HASH_KEY_IP,
+    H2O_BALANCER_HASH_KEY_IP_PORT,
+    H2O_BALANCER_HASH_KEY_PATH
+} h2o_balancer_hash_key_type_t;
```

Comment:
Am I correct in assuming that this is an internal of the configurator? If that is the case, can we move them into the configurator?